modules/text2vec-openai/vectorizer/objects.go

```diff
@@ -21,21 +26,35 @@ import (
 	libvectorizer "github.com/weaviate/weaviate/usecases/vectorizer"
 )
 
+type batchJob struct {
+	objects []*models.Object
+	texts   []string
+	tokens  []int
+	ctx     context.Context
+	wg      *sync.WaitGroup
+	errs    map[int]error
+	cfg     moduletools.ClassConfig
+}
+
 type Vectorizer struct {
 	client           Client
 	objectVectorizer *objectsvectorizer.ObjectVectorizer
+	jobQueueCh       chan batchJob
 }
 
 func New(client Client) *Vectorizer {
-	return &Vectorizer{
+	vec := &Vectorizer{
 		client:           client,
 		objectVectorizer: objectsvectorizer.New(),
+		jobQueueCh:       make(chan batchJob),
 	}
+
+	go vec.batchWorker()
```

Comment:
wouldn't it be better to run the vectorization job only when the batch request comes in? it's another goroutine that works in the background and will be only used when a batch request comes in bc it's not used for objects and also not for texts, so I assume that it would be idle for 90% of time.

Reply:
How would this worked work if there would come n batch requests? bc with named vectors it will be a possibility to define n OpenAI vectorizers with n configurations

Reply:
> How would this worked work if there would come n batch requests? bc with named vectors it will be a possibility to define n OpenAI vectorizers with n configurations

same as now where you could have different classes with different vectorizer configs. In `ObjectBatch` each batch is tokenized according to its model and the config is send as part of the job. With named vectors we'd need one batch-batch job per vectorizer, eg if we have 1000 objects coming in and 3 named vectors, we could call `ObjectBatch` for each vectorizer

Reply:
>  wouldn't it be better to run the vectorization job only when the batch request comes in? it's another goroutine that works in the background and will be only used when a batch request comes in bc it's not used for objects and also not for texts, so I assume that it would be idle for 90% of time.

hmmm we could start the background job when batches come in and stop it if the last batch is done, same as in the repo-worker queue 🤔

Reply:
Just checked the repo-worker queue and startup/shutdown on demand is not there anymore - so there a few goroutines running/idling for all the time.

Is one more really a problem? AFAIK they take almost no resources when blocking on a channel receive